examples/train_model.py

```diff
@@ -93,13 +94,20 @@ def main():
     train.add_argument('-dbf', '--dict-build-first',
                         type='bool', default=True,
                         help='build dictionary first before training agent')
+    train.add_argument('--tensorboard-dir', default='',
+                        help='directory for tensorboard logging')
     opt = parser.parse_args()
     # Possibly build a dictionary (not all models do this).
     if opt['dict_build_first'] and 'dict_file' in opt:
         if opt['dict_file'] is None and opt.get('model_file'):
             opt['dict_file'] = opt['model_file'] + '.dict'
         print("[ building dictionary first... ]")
         build_dict.build_dict(opt)
+
+    # Set up logger if directory provided
+    if opt['tensorboard_dir']:
+        opt['logger'] = TensorboardLogger(opt['tensorboard_dir'])
```

Comment:
we are trying to keep opt as only strings, float, bools etc,  not whole classes.